plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -216,6 +221,22 @@
     private final List<String> tableTypes;
     private final AggregateFunctionRewriter aggregateFunctionRewriter;
 
+    private static final PredicatePushdownController POSTGRESQL_CHARACTER_PUSHDOWN = (session, domain) -> {
+        checkArgument(
+                domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
+                "This PredicatePushdownController can be used only for chars and varchars");
+
+        if (domain.isOnlyNull() ||
+                // PostgreSQL is case sensitive by default
+                domain.getValues().isDiscreteSet()) {
```

Comment:
I'd put these two conditions on one line, with the comment inside the `if` block.

Maybe also add a bit more to the comment explaining how `isDiscreteSet` is related to case sensitivity. The comment doesn't really clarify much right now. (Ideally, I think `CASE_INSENSITIVE_CHARACTER_PUSHDOWN` would have a more detailed explanation of how that works, and then we could reference it here.)

Reply:
I put the comment in between so it unambiguous what it refers to

point take on the issue how discrete sets translate to actual predicates, and indeed it applies to CASE_INSENSITIVE_CHARACTER_PUSHDOWN